src/parallel/DomainDecomposition.h

```diff
@@ -23,6 +23,7 @@ class DomainDecomposition : public DomainDecompMPIBase {
 	//!        sets up the topology
 	DomainDecomposition();
 
+	DomainDecomposition(MPI_Comm comm, std::array<int, DIMgeom> gridSize);
```

Comment:
```suggestion
	DomainDecomposition(MPI_Comm comm, const std::array<int, DIMgeom> &gridSize);
```
don't create unnecessary copies